lib/grape_entity/entity.rb

```diff
@@ -391,9 +391,17 @@ def self.present_collection(present_collection = false, collection_name = :items
     # @option options :only [Array] all the fields that should be returned
     # @option options :except [Array] all the fields that should not be returned
     def self.represent(objects, options = {})
-      if objects.respond_to?(:to_ary) && ! @present_collection
-        root_element =  root_element(:collection_root)
-        inner = objects.to_ary.map { |object| new(object, options.reverse_merge(collection: true)).presented }
+      collection_method_name = if objects.respond_to?(:to_ary)
+        :to_ary
+      elsif ! objects.nil? && ! objects.is_a?(Hash) && objects.respond_to?(:to_a)
+        :to_a
+      end
+
+      if collection_method_name && ! @present_collection
+        root_element = root_element(:collection_root)
+        inner = objects.send(collection_method_name).map do |object|
+          new(object, options.reverse_merge(collection: true)).presented
+        end
       else
         objects = { @collection_name => objects } if @present_collection
```

Comment:
I think storing the method name is strange, or at least not idiomatic. How's this?

```ruby
objects_collection = if objects.respond_to?(:to_ary)
  objects.to_ary
elsif ! objects.is_a?(Hash) && objects.respond_to?(:to_a) #nil? is redundant here
  objects.to_a
end

if objects_collection && !@present_collection
  root_element = root_element(:collection_root)
  inner = objects_collection.map ...
elsif @present_collection
  objects = { @collection_name => objects }
end
```